internal/log/log.go

```diff
@@ -99,10 +99,21 @@ var (
 )
 
 func init() {
-	if v := os.Getenv("DD_LOGGING_RATE"); v != "" {
-		if sec, err := strconv.ParseUint(v, 10, 64); err != nil {
-			Warn("Invalid value for DD_LOGGING_RATE: %v", err)
+	v := os.Getenv("DD_LOGGING_RATE")
+	setLoggingRate(v)
+}
+
+func setLoggingRate(v string) {
+	if v == "" {
+		return
+	}
+	if sec, err := strconv.ParseInt(v, 10, 64); err != nil {
+		Warn("Invalid value for DD_LOGGING_RATE: %v", err)
+	} else {
+		if sec < 0 {
+			Warn("Invalid value for DD_LOGGING_RATE: negative value")
 		} else {
+			// DD_LOGGING_RATE = 0 allows to log errors immediately.
```

Comment:
Do you think this is intentional behavior, or a bug? There are zero docs for `DD_LOGGING_RATE` so perhaps this is just intended for internal use...?

Reply:
@katiehockman I found this as a comment in [a issue from 2020](https://github.com/DataDog/dd-trace-go/issues/650#issuecomment-623288844).

It seems to be intentional but undocumented.